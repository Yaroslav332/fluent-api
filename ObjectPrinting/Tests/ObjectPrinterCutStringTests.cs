using ApprovalTests;
using ApprovalTests.Reporters;
using NUnit.Framework;

namespace ObjectPrinting.Tests
{
    [UseReporter(typeof(DiffReporter))]
    [TestFixture]
    public class ObjectPrinterCutStringTests
    {
        [Test]
        public void CutString_Should_CutCorrectly_WhenStringIsLarge()
        {
            var person = new Person {Name = "Alexander", Age = 19};
            var printer = ObjectPrinter.For<Person>()
                .ConfigForProperty(p => p.Name)
                .CutString(4);
            Approvals.Verify(printer.PrintToString(person));
        }

        [Test]
        public void CutString_ShouldNot_Cut_WhenStringIsSmall()
        {
            var person = new Person {Name = "Al", Age = 19};
            var printer = ObjectPrinter.For<Person>()
                .ConfigForProperty(p => p.Name)
                .CutString(4);
            Approvals.Verify(printer.PrintToString(person));
        }

        [Test]
        public void CutString_ShouldNot_Cut_WhenStringIsEqualToMaxLength()
        {
            var person = new Person {Name = "Alex", Age = 19};
            var printer = ObjectPrinter.For<Person>()
                .ConfigForProperty(p => p.Name)
                .CutString(4);
            Approvals.Verify(printer.PrintToString(person));
        }

        [Test]
        public void CutString_ShouldNot_Override_OtherSerializeMethod()
        {
            var person = new Person {Name = "Alexander", Age = 19};
            var printer = ObjectPrinter.For<Person>()
                .ConfigForProperty(p => p.Name)
                .UseSerializeMethod(s => "Call me " + s)
                .ConfigForProperty(p => p.Name)
                .CutString(4);
            Approvals.Verify(printer.PrintToString(person));
        }

        [Test]
        public void OverrideStringSerializeMethod_ShouldNot_Override_CutString()
        {
            var person = new Person {Name = "Alexander", Age = 19};
            var printer = ObjectPrinter.For<Person>()
                .ConfigForProperty(p => p.Name)
                .CutString(4)
                .ConfigForProperty(p => p.Name)
                .UseSerializeMethod(s => "Call me " + s);
            Approvals.Verify(printer.PrintToString(person));
        }

        [Test]
        public void CutString_Should_Override_PrevCutString()
        {
            var person = new Person {Name = "Alexander", Age = 19};
            var printer = ObjectPrinter.For<Person>()
                .ConfigForProperty(p => p.Name)
                .CutString(2)
                .ConfigForProperty(p => p.Name)
                .CutString(4);
            Approvals.Verify(printer.PrintToString(person));
        }
    }
}