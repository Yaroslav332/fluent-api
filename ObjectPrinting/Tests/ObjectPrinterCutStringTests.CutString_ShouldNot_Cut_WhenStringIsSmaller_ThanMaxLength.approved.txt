Person
	Id = 00000000-0000-0000-0000-000000000000
	Name = Al
	Height = 0
	Weight = 0
	Birthday = 01.01.0001 0:00:00
	RunRatio = 0
	Age = 19
	RunSpeed = 0
	Parent = null
